main.go

```diff
@@ -480,20 +481,44 @@ func (c *config) format(file ast.Node, rwErrs error) (string, error) {
 		// back. Second, our cursor might intersect two different structs with
 		// other declarations in between them. Printing the file and cutting
 		// the selection is the easier and simpler to do.
-		var buf bytes.Buffer
-		err := format.Node(&buf, c.fset, file)
+		var src bytes.Buffer
+
+		// this is the default config from `format.Node()`, but we add
+		// `printer.SourcePos` to get the original source position of the
+		// modified lines
+		cfg := printer.Config{Mode: printer.SourcePos | printer.UseSpaces | printer.TabIndent, Tabwidth: 8}
+		err := cfg.Fprint(&src, c.fset, file)
 		if err != nil {
 			return "", err
 		}
 
+		scanner := bufio.NewScanner(&src)
+
 		var lines []string
-		scanner := bufio.NewScanner(bytes.NewBufferString(buf.String()))
 		for scanner.Scan() {
-			lines = append(lines, scanner.Text())
+			txt := scanner.Text()
+
+			// check for any line directive and store it for next iteration to
+			// re-construct the original file
+			if !strings.HasPrefix(txt, "//line") {
+				lines = append(lines, txt)
+				continue
+			}
+
+			line := strings.Split(txt, ":")
+			lineNr, err := strconv.Atoi(line[1])
```

Comment:
This will panic if it looks like this:

```
//line blabla.txt
```

(Don’t know if that can happen)

Reply:
Yeah! I fixed it locally but forget to push it. Thanks for reminding :)